### PR TITLE
Feature/sc 4045 registration without consent

### PR DESCRIPTION
**Pull request by pkleybolte**
# Description

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-4045

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5524126&bid=18100129)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5524126&bid=18100129)

File: static/scripts/dataprivacy/paginated_form.js [hunk 1]
```diff
@@ -102,11 +103,15 @@ function setSelectionByIndex(index, event) {
 				sectionIndex: getSelectionIndex(),
 			},
 		});
-		document.querySelector(`section[data-panel="section-${getSelectionIndex()}"]`).dispatchEvent(hideEvent);
+		console.log(document.querySelectorAll('form .panels > section[data-panel]'));
```
Comment (CeEv) [inline static/scripts/dataprivacy/paginated_form.js:106]: Please remove console.logs

File: static/scripts/dataprivacy/paginated_form.js [hunk 2]
```diff
@@ -151,6 +161,12 @@ function setSelectionByIndex(index, event) {
 	}
 }
 
+function skipConsent() {
+	const skipConsentFor = 'student'; // TODO get from feature toggle
```
Comment (CeEv) [inline static/scripts/dataprivacy/paginated_form.js:165]: open todo?

File: static/scripts/dataprivacy/paginated_form.js [hunk 3]
```diff
@@ -233,6 +249,13 @@ window.addEventListener('DOMContentLoaded', () => {
 	if (nextButton) {
 		nextButton.addEventListener('click', nextSection);
 	}
+
+	// if (skipConsent()) {
```
Comment (CeEv) [inline static/scripts/dataprivacy/paginated_form.js:253]: It is not needed, then please remove.

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5524126&bid=18100129)

File: static/scripts/dataprivacy/paginated_form.js (hunk 2 above)
Comment (pkleybolte) [reply static/scripts/dataprivacy/paginated_form.js]: Actually the function is not needed. I removed it completly.

File: static/scripts/dataprivacy/paginated_form.js (hunk 1 above)
Comment (pkleybolte) [reply static/scripts/dataprivacy/paginated_form.js]: Done

File: static/scripts/dataprivacy/paginated_form.js (hunk 3 above)
Comment (pkleybolte) [reply static/scripts/dataprivacy/paginated_form.js]: done

**Pull request merged**